sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/ShuffleExchangeExec.scala

```diff
@@ -92,10 +92,17 @@ case object REPARTITION_BY_COL extends ShuffleOrigin
 // a certain partition number. Spark can't optimize it.
 case object REPARTITION_BY_NUM extends ShuffleOrigin
 
-// Indicates that the shuffle operator was added by the user-specified repartition operator. Spark
-// firstly tries to coalesce partitions, if it cannot be coalesced, then use the local shuffle
-// reader.
-case object REPARTITION_BY_NONE extends ShuffleOrigin
+// Indicates that the rebalance operator was added by the user-specified repartition operator.
```

Comment:
the shuffle operator was added by the user-specified rebalance operator.

Reply:
same for https://github.com/apache/spark/pull/32932/files#diff-7ecb26a3ce3210f09ab99080d3313aaaf7f6ffa4bda4f80207a686dd812d3a1bR100